client/secureclient/src/main/java/com/linkedin/openhouse/client/ssl/WebClientFactory.java

```diff
@@ -154,14 +154,16 @@ private WebClient createSecureWebClient(String baseUrl, String truststoreLocatio
               .secure(t -> t.sslContext(createSslContext(truststore)));
     }
 
-    ClientHttpConnector connector = new ReactorClientHttpConnector(client);
+    return client;
+  }
+
+  private WebClient.Builder getWebClientBuilder(String baseUrl) {
     WebClient.Builder webClientBuilder = createWebClientBuilder();
     setSessionIdInWebClientHeader(webClientBuilder);
+    setClientNameInWebClientHeader(webClientBuilder);
     return webClientBuilder
-        .baseUrl(baseUrl)
-        .clientConnector(connector)
```

Comment:
I think we can set the connector as well here and the builder initialization will be completed in one place together and we can directly return webClient.

Reply:
fair point, fixed this